src/typesystem.cpp

```diff
@@ -6,6 +6,56 @@
 
 #include "compile_time_environment.hpp"
 
+TypeSystemCore::TypeSystemCore() {
+	m_tf_core.unify_function = [&](int a, int b) {
+		if (m_tf_core.find_term(a) == m_tf_core.find_term(b))
+			return;
+	
+		TypeFunctionData& a_data = m_type_functions[m_tf_core.find_function(a)];
+		TypeFunctionData& b_data = m_type_functions[m_tf_core.find_function(b)];
+
+		if (b_data.is_dummy) {
+			std::swap(a, b);
+			std::swap(a_data, b_data);
+		}
+
+		if (a_data.is_dummy) {
+			for (auto& kv_a : a_data.structure) {
+				auto kv_b = b_data.structure.find(kv_a.first);
+
+				if (kv_b == b_data.structure.end())
```

Comment:
Leaving single comment on random line

Reply:
Okay. This *has* to be non-intentional. There's clearly a "single comment" and a "review" button. It must be a bug.